src/lib/bootloader/proposal_client.rb

```diff
@@ -36,9 +36,16 @@ def initialize
     def make_proposal(attrs)
       force_reset = attrs["force_reset"]
       auto_mode = Yast::Mode.autoinst || Yast::Mode.autoupgrade
-
-      if (force_reset || !Yast::Bootloader.proposed_cfg_changed) &&
-          !auto_mode
+      storage_changed = Yast::BootStorage.storage_changed?
+      # redetect disks if cache is invalid as first part
+      Yast::BootStorage.detect_disks if storage_changed
+
+      # logic in this condition:
+      # when reset is forced or user do not modify proposal, reset proposal,
+      # but only when not using auto_mode
+      # But if storage changed, always repropose as it can be very wrong.
+      if ((force_reset || !Yast::Bootloader.proposed_cfg_changed) &&
+          !auto_mode) || storage_changed
```

Comment:
NP: maybe moving this to a method would allow more clear conditions, like (just a possible example):
```ruby
def reset_needed?(force_reset)
  return true if storage_changed
  return false if Yast::Mode.autoinst || Yast::Mode.autoupgrade
  force_reset || !Yast::Bootloader.proposed_cfg_changed
end
```